src/bsp/src/stm32/hal/src/hal_flash.c

```diff
@@ -0,0 +1,37 @@
+#include "hal/hal_flash.h"
+
+bool Flash_eraseSector(uint8_t sector) {
+    FLASH_EraseInitTypeDef eraseConfig = {.TypeErase = FLASH_TYPEERASE_SECTORS,
+                                          .Sector = sector,
+                                          .NbSectors = 1,
+                                          .VoltageRange = FLASH_VOLTAGE_RANGE_3};
+    uint32_t badSector = 0;
+
+    HAL_FLASH_Unlock();
+    bool ret = HAL_FLASHEx_Erase(&eraseConfig, &badSector) == HAL_OK;
+    HAL_FLASH_Lock();
+
+    return ret;
+}
+
+bool Flash_program(uint32_t address, uint8_t* data, uint32_t size) {
+    if (data == NULL) {
+        return false;
+    }
+
+    HAL_FLASH_Unlock();
+
+    for (unsigned int i = 0; i < size; i++) {
```

Comment:
Should we check that the address is in the user data bounds? Or maybe a wrapper that does this check to keep the logic of this class of only writing to flash. (So we could have one wrapper for buzz code and one for settings to make sure both have their separate spaces)